packages/i18n/src/t/translation-binding.ts

```diff
@@ -0,0 +1,254 @@
+    if (typeof newValue === 'object') {
+      this.translationParameters = newValue;
+    } else {
+      this.keyExpression = this.isInterpolatedSourceExpr
+        ? this.expr.evaluate(flags, this.scope, this.locator, '') as string
+        : newValue;
+    }
+    this.updateTranslations(flags);
+  }
+
+  private handleLocaleChange() {
+    this.updateTranslations(LifecycleFlags.none);
+  }
+
+  private updateTranslations(flags: LifecycleFlags) {
+    const results = this.i18n.evaluate(this.keyExpression, this.translationParameters);
+    const content: ContentValue = Object.create(null);
+    this.unobserveTargets(flags);
+
+    for (const item of results) {
+      const value = item.value;
+      const attributes = this.preprocessAttributes(item.attributes);
+      for (const attribute of attributes) {
+        if (this.isContentAttribute(attribute)) {
+          content[attribute] = value;
+        } else {
+          this.updateAttribute(attribute, value, flags);
+        }
+      }
+    }
+    if (Object.keys(content).length) {
+      this.updateContent(content, flags);
+    }
+  }
+
+  private updateAttribute(attribute: string, value: string, flags: LifecycleFlags) {
+    const controller = CustomElement.behaviorFor(this.target);
+    const observer = controller && controller.viewModel
+      ? this.observerLocator.getAccessor(LifecycleFlags.none, controller.viewModel, attribute)
+      : this.observerLocator.getAccessor(LifecycleFlags.none, this.target, attribute);
+    observer.setValue(value, flags);
+    this.targetObservers.add(observer);
+  }
+
+  private preprocessAttributes(attributes: string[]) {
+    if (attributes.length === 0) {
+      attributes = this.target.tagName === 'IMG' ? ['src'] : ['textContent'];
+    }
+    const htmlIndex = attributes.findIndex((attr) => attr === 'html');
+    if (htmlIndex > -1) {
+      attributes.splice(htmlIndex, 1, 'innerHTML');
+    }
+    return attributes;
+  }
+
+  private isContentAttribute(attribute: string): attribute is ContentAttribute {
+    return this.contentAttributes.includes(attribute);
+  }
+
+  private updateContent(content: ContentValue, flags: LifecycleFlags) {
+    const children = Array.from(this.target.childNodes);
+    const fallBackContents = [];
+    const marker = 'au-i18n';
+
+    // extract the original content, not manipulated by au-i18n
+    for (const child of children) {
+      if (!Reflect.get(child, marker)) {
+        fallBackContents.push(child);
+      }
+    }
+
+    const template = this.prepareTemplate(content, marker, fallBackContents);
+
+    // difficult to use the set property approach in this case, as most of the properties of Node is readonly
+    // const observer = this.observerLocator.getAccessor(LifecycleFlags.none, this.target, '??');
+    // observer.setValue(??, flags);
+
+    this.target.innerHTML = '';
+    for (const child of Array.from(template.content.childNodes)) {
```

Comment:
```suggestion
    for (const child of PLATFORM.toArray(template.content.childNodes)) {
```

Is about 10x faster than `Array.from` :)

Reply:
Done